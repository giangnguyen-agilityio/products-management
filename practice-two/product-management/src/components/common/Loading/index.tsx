// Libraries
import { memo } from 'react'
import { Container, Flex, Spinner, Text } from '@chakra-ui/react'

const Loading = () => {
  return (
    <Container>
      <Flex
        justifyContent="center"
        flexDirection="column"
        margin="0 auto"
        height="100vh"
        borderRadius={10}
        fontFamily="Oswald-Regular"
        alignItems="center"
      >
        {/* Display a spinning loading indicator */}
        <Spinner
          thickness="4px"
          speed="0.65s"
          emptyColor="gray.200"
          size="xl"
        />
        {/* Display a text indicating the loading process */}
        <Text
          variant="fontSecondary"
          fontSize="lg"
          color="secondary"
          marginTop={2}
        >
          Loading....
        </Text>
      </Flex>
    </Container>
  )
}

export default memo(Loading)
