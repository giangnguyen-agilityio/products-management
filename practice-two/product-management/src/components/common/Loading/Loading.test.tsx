import { render } from '@testing-library/react'
import Loading from './'

describe('Loading Component', () => {
  it('should render the loading spinner', () => {
    const { container } = render(<Loading />)
    expect(container).toMatchSnapshot()
  })
})
