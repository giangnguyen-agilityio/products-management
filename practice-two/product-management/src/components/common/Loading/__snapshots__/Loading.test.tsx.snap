// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should render the loading spinner 1`] = `
<div>
  <div
    class="chakra-container css-0"
  >
    <div
      class="css-gv2u4r"
    >
      <div
        class="chakra-spinner css-mb5q2b"
      >
        <span
          class="css-8b45rq"
        >
          Loading...
        </span>
      </div>
      <p
        class="chakra-text css-xyfjrl"
      >
        Loading....
      </p>
    </div>
  </div>
</div>
`;
